R/patch.R

```diff
@@ -368,6 +373,7 @@ tbl_patch <- function(snapshot, update, ukey_names) {
   result_tbl <- vec_rbind(update, snapshot)
 
   dup_ids <- vec_duplicate_id(result_tbl[ukey_names])
+  # check that the index hasn't be reset to something lower that it duplicates
```

Comment:
```suggestion
  # Find the "first" appearance of each ukey; since `update` is ordered before `snapshot`,
  # this means favoring the rows from `update` over those in `snapshot`.
  # This is like `!duplicated()` but faster, and like `vec_unique_loc()` but guaranteeing
  # that we get the first appearance since `vec_duplicate_id()` guarantees that
  # it points to the first appearance.
```